packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -404,12 +404,35 @@ export class FieldSchemaAlpha<
 	}
 }
 
+/**
+ * {@link FieldSchemaAlpha} including {@link SimpleObjectFieldSchema}.
+ */
+export class ObjectFieldSchema<
+		Kind extends FieldKind = FieldKind,
+		Types extends ImplicitAllowedTypes = ImplicitAllowedTypes,
+		TCustomMetadata = unknown,
+	>
+	extends FieldSchemaAlpha<Kind, Types, TCustomMetadata>
+	implements SimpleObjectFieldSchema
+{
+	public readonly storedKey: string;
+
+	public constructor(
+		kind: Kind,
+		allowedTypes: Types,
+		props: FieldProps<TCustomMetadata> & { readonly key: string },
+	) {
+		super(kind, allowedTypes, props);
+		this.storedKey = props.key;
+	}
+}
+
 /**
  * Normalizes a {@link ImplicitFieldSchema} to a {@link FieldSchema}.
  */
-export function normalizeFieldSchema(schema: ImplicitFieldSchema): FieldSchema {
+export function normalizeFieldSchema(schema: ImplicitFieldSchema): FieldSchemaAlpha {
 	return schema instanceof FieldSchema
-		? schema
+		? (schema as FieldSchemaAlpha)
```

Comment:
Should we be `instanceof` checking against `FieldSchemaAlpha` now?

Reply:
Tried that. Since the type checker doesn't know all field schema are FieldSchemaAlpha, that forces a different typer cast that is less clearly safe below. Adding a second instance of and an assert here is possible, but given the constructor for FieldSchema enforces this already, it seemed unnecessary.